agent/consul/leader.go

```diff
@@ -448,14 +448,6 @@ func (s *Server) initializeCA() error {
 		return err
 	}
 
-	// TODO(banks): in the case that we've just gained leadership in an already
-	// configured cluster. We really need to fetch RootCA from state to provide it
-	// in setCAProvider. This matters because if the current active root has
-	// intermediates, parsing the rootCA from only the root cert PEM above will
-	// not include them and so leafs we sign will not bundle the intermediates.
-
-	s.setCAProvider(provider, rootCA)
-
```

Comment:
I think this breaks things.

The whole reason for this thing being an issue IIRC was that we had to set this before the `activeRoot` check.

`activeRoot != nil` _only_ happens during initial bootstrap, so moving `setCAProvider` to there means that the intermediates are not bundled and so are not served with the leaf.

What am I missing?